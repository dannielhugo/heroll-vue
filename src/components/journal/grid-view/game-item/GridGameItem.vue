<template>
  <el-card class="game-card" :body-style="{ padding: '0px' }" shadow="hover">
    <el-image :src="game.background_image" fit="contain" />
    <div style="padding: 14px">
      <div class="game-card__title">{{ game.name }}</div>
      <div class="game-card__subtitle">{{ genres }}</div>
      <div class="game-card__subtitle">{{ platforms }}</div>
    </div>
  </el-card>
</template>

<script setup lang="ts">
import type { Game } from '@/models/game';

const props = defineProps<{ game: Game }>();

const platforms = props.game.platforms?.reduce((curr, game) => {
  if (curr === '') return game.platform.name;
  return `${curr}, ${game.platform.name}`;
}, '');

const genres = props.game.genres?.reduce((curr, genre) => {
  if (curr === '') return genre.name;
  return `${curr}, ${genre.name}`;
}, '');
</script>

<style scoped lang="scss">
.game-card {
  width: 345px;
  cursor: pointer;

  &__image {
    width: 100%;
  }

  &__title {
    font-weight: 400;
    font-size: 1.5rem;
  }

  &__subtitle {
    font-weight: 400;
    font-size: 0.875rem;
    margin-top: 2rem;
  }
}
</style>
