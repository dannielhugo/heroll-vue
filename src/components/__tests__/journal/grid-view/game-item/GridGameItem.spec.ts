import { mount } from '@vue/test-utils';
import ElementPlus from 'element-plus';
import GridGameItem from '../../../../journal/grid-view/game-item/GridGameItem.vue';
import * as game from '../../../mocks/Game.json';

describe('GridGameItem', () => {
  test('should mount component', async () => {
    expect(GridGameItem).toBeTruthy();

    const wrapper = mount(GridGameItem, {
      props: {
        game,
      },
      global: {
        plugins: [ElementPlus],
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
