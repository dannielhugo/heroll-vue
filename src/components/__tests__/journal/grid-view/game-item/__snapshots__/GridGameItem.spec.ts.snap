// Vitest Snapshot v1

exports[`GridGameItem > should mount component 1`] = `
"<div class=\\"el-card is-hover-shadow game-card\\" data-v-a56eedd2=\\"\\">
  <!--v-if-->
  <div class=\\"el-card__body\\" style=\\"padding: 0px;\\">
    <!--v-if-->
    <div style=\\"padding: 14px;\\" data-v-a56eedd2=\\"\\">
      <div class=\\"game-card__title\\" data-v-a56eedd2=\\"\\">Soulcalibur (1998)</div>
      <div class=\\"game-card__subtitle\\" data-v-a56eedd2=\\"\\">Fighting</div>
      <div class=\\"game-card__subtitle\\" data-v-a56eedd2=\\"\\">Xbox 360, Dreamcast</div>
    </div>
  </div>
</div>"
`;
